pkg/network/ebpf/c/protocols/tls/sowatcher.h

```diff
@@ -54,11 +54,20 @@ static __always_inline void do_sys_open_helper_exit(exit_sys_openat_ctx *args) {
         return;
     }
 
-    // Detect whether the file being opened is a shared library
+// Check the last 6 characters of the following libraries to ensure the file is a relevant `.so`.
+// Libraries:
+//    libssl.so -> ssl.so
+// libcrypto.so -> pto.so
+// libgnutls.so -> tls.so
     bool is_shared_library = false;
+#define match2chars(_base, _a,_b) (path->buf[_base+i] == _a && path->buf[_base+i+1] == _b)
+#define match3chars(_base, _a,_b,_c) (path->buf[_base+i] == _a && path->buf[_base+i+1] == _b && path->buf[_base+i+2] == _c)
 #pragma unroll
-    for (int i = 0; i < LIB_PATH_MAX_SIZE - SO_SUFFIX_SIZE; i++) {
-        if (path->buf[i] == '.' && path->buf[i + 1] == 's' && path->buf[i + 2] == 'o') {
+    for (int i = 0; i < LIB_PATH_MAX_SIZE - (LIB_SO_SUFFIX_SIZE); i++) {
+        if ((match2chars(0, 's','l') ||
+                match2chars(0, 't','o') ||
+                match2chars(0, 'l','s')) &&
+            match3chars(3, '.','s','o')) {
```

Comment:
Can you elaborate why did change to comparison of the last 5 characters?

Reply:
6 matching chars with 2 subset pattern generate too much complexity for some verifier (4.14 / 5.10)

I trying different strategy, but probably the slightly more matching (less complexity) previous filter (ssl pto tls sts pss ...) may be an alternative